seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/source/KafkaSourceSplitEnumerator.java

```diff
@@ -180,7 +182,6 @@ public void close() throws IOException {
     public void addSplitsBack(List<KafkaSourceSplit> splits, int subtaskId) {
         if (!splits.isEmpty()) {
             pendingSplit.putAll(convertToNextSplit(splits));
-            assignSplit();
```

Comment:
During runtime, if new partition are discovered, will they not be distributed anymore?

Reply:
`addSplitsBack()` shouldn't do the assign split, it lags behind `registeredReaders()`. In addition, run() will assign split again and it is run in the next stage.

Reply:
![image](https://github.com/apache/incubator-seatunnel/assets/48236177/f4a194f9-40da-45d6-a069-82d578346201)

Reply:
During normal operation, when a topic generates new partitions, we can automatically discover them and distribute to the reader for reading.

Reply:
I misread.